code/game/machinery/tcomms/core.dm

```diff
@@ -257,9 +266,35 @@
 			to_chat(usr, "<span class='notice'>Successfully changed password from <b>[link_password]</b> to <b>[new_password]</b>.</span>")
 			link_password = new_password
 
+	if(ui_tab == UI_TAB_FILTER)
+		if(href_list["add_filter"])
+			// This is a stripped input because I did NOT come this far for this system to be abused by HTML injection
+			var/name_to_add = stripped_input(usr, "Enter a name to add to the filtering list", "Name Entry")
+			if(name_to_add == "")
+				return
+			if(name_to_add in nttc.filtering)
```

Comment:
Performance wise it would be better to use an assoc list. Where the key is the name and the value is just 1. O(1) complexity VS O(n) complexity.

Reply:
Considering this list is going to rarely be used, and even when it is, it will have about 2 people inside it, I dont see this as being a huge issue, and I would prefer this for readability.